### PR TITLE
Add arm tests to mps workflow

### PR DESCRIPTION
**Pull request by clee2000**
Fixes #ISSUE_NUMBER
https://github.com/pytorch/pytorch/pull/94417/files broke mac tests but was an mps pr, so add arm tests to mps

### <samp>🤖 Generated by Copilot at 4b0f4ed</samp>

> _`PyTorch` on `macOS`_
> _Testing new Python and ARM_
> _Autumn of Silicon_

### CODE REVIEW

File: .github/workflows/mac-mps.yml [hunk 1]
```diff
@@ -33,6 +33,18 @@ jobs:
       MACOS_SCCACHE_S3_ACCESS_KEY_ID: ${{ secrets.MACOS_SCCACHE_S3_ACCESS_KEY_ID }}
       MACOS_SCCACHE_S3_SECRET_ACCESS_KEY: ${{ secrets.MACOS_SCCACHE_S3_SECRET_ACCESS_KEY }}
 
+  macos-12-py3-arm64-test:
```
Comment (huydhn) [inline .github/workflows/mac-mps.yml:36]: I'm a bit confuse looking at this.  Does this make `mac-mps` workflow look exactly like `trunk` when it comes to MacOS testing?  In that case, would that be easier to have an auto label rule to add `ciflow/trunk` for every mps-related PR instead?

Comment (clee2000) [reply .github/workflows/mac-mps.yml:36]: I think the idea is that all of trunk takes too long and people working on mps tend to merge as soon as mps is finished

I'm honestly not too sure how necessary this is given that it's only happened once, so if people have thoughts on whether this should happen, I'm open to it

Comment (ZainRizvi) [reply .github/workflows/mac-mps.yml:36]: The tradeoff here sounds like either making this mac-mps workflow a bit more complex (with some code duplication) vs running trunk more often on mac workflows (causing more jobs to be run and potentially some job duplication)

Given the focus on capacity constraints and not wanting to use more infrastructure than needed, I'd lean towards adding an extra job in this mac-mps workflow than adding ciflow/trunk to all mps builds

**Review by ZainRizvi — APPROVED** [review]

**Comment by kit1980** [comment]
copilot:poem

**Comment by malfet** [comment]
To be frank, this feels like a huge overkill. I.e. PR used as an example, was not a pure MPS changes, so developer were supposed to wait for the trunk jobs to finish, rather than `merge -f`. So, I propose we revert this change to keep MPS jobs really small.

**Pull request closed without merging**